### ISSUE TITLE
2.0.0 fails to build on FreeBSD with Clang >= 15

### ISSUE THREAD

**Issue by jbeich** [project-repo/cagebreak#36]
[reallocarray(3)](https://man.freebsd.org/reallocarray/3) is not in POSIX thus not visible with `_POSIX_C_SOURCE`. Linux cheats POSIX via `_DEFAULT_SOURCE` but on BSDs the cheat knob is different (`__BSD_VISIBLE` on DragonFly/FreeBSD/OpenBSD and `_NETBSD_SOURCE` on NetBSD).

```
$ cc --version
FreeBSD clang version 15.0.6 (https://github.com/llvm/llvm-project.git llvmorg-15.0.6-0-g088f33605d8a)
Target: x86_64-unknown-freebsd14.0
Thread model: posix
InstalledDir: /usr/bin
```

```c
$ meson setup _build
$ meson compile -C _build
[...]
../cagebreak.c:197:11: warning: call to undeclared function 'reallocarray'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
                        line = reallocarray(line, line_length, sizeof(char));
                               ^
../cagebreak.c:197:9: error: incompatible integer to pointer conversion assigning to 'char *' from 'int' [-Wint-conversion]
                        line = reallocarray(line, line_length, sizeof(char));
                             ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Note, on Clang < 15 `-Wint-conversion` doesn't error out, so the implicit declaration of `reallocarray` sometimes leads to crashes.

**Comment by jbeich** [comment]
The common way to fix such issues is to drop `_POSIX_C_SOURCE` (or `_XOPEN_SOURCE`). If it's not tested on CI there's no point pretending to be POSIX compliant.

**Comment by jbeich** [comment]
Fixed in 2.0.1.
